plugins/modules/azure_rm_afdendpoint_info.py

```diff
@@ -0,0 +1,272 @@
+        location:
+            description:
+                - Location of the AFD Endpoint.
+            type: str
+            sample: Global
+        name:
+            description:
+                - Name of the AFD Endpoint.
+            returned: always
+            type: str
+            sample: myEndpoint
+        profile_name:
+            description:
+                - Name of the AFD Profile which holds the Endpoint.
+            returned: always
+            type: str
+            sample: myProfile
+        provisioning_state:
+            description:
+                - Provisioning status of the AFD Endpoint.
+            type: str
+            sample: Succeeded
+        resource_group:
+            description:
+                - Name of a resource group where the AFD Endpoint exists.
+            returned: always
+            type: str
+            sample: myResourceGroup
+        tags:
+            description:
+                - The tags of the AFD Endpoint.
+            type: list
+            sample: foo
+        type:
+            description:
+                - Resource type.
+            type: str
+            sample: Microsoft.Cdn/profiles/afdendpoints
+'''
+
+from ansible_collections.azure.azcollection.plugins.module_utils.azure_rm_common import AzureRMModuleBase
+
+try:
+    from azure.mgmt.cdn import CdnManagementClient
+except ImportError:
+    # handled in azure_rm_common
+    pass
+
+import re
+
+AZURE_OBJECT_CLASS = 'AFDEndpoint'
+
+
+class AzureRMAFDEndpointInfo(AzureRMModuleBase):
+    """Utility class to get Azure AFD Endpoint facts"""
+
+    def __init__(self):
+
+        self.module_args = dict(
+            name=dict(type='str'),
+            resource_group=dict(
+                type='str',
+                required=True
+            ),
+            profile_name=dict(
+                type='str',
+                required=True
+            ),
+            tags=dict(
+                type='list',
+                elements='str'
+            )
+        )
+
+        self.results = dict(
+            changed=False,
+            afdendpoints=[]
+        )
+
+        self.name = None
+        self.resource_group = None
+        self.profile_name = None
+        self.tags = None
+
+        super(AzureRMAFDEndpointInfo, self).__init__(
+            supports_check_mode=True,
+            derived_arg_spec=self.module_args,
+            supports_tags=False,
+            facts_module=True
+        )
+
+    def exec_module(self, **kwargs):
+
+        for key in self.module_args:
+            setattr(self, key, kwargs[key])
+
+        self.endpoint_client = self.get_mgmt_svc_client(CdnManagementClient,
+                                                   base_url=self._cloud_environment.endpoints.resource_manager,
+                                                   api_version='2023-05-01')
```

Comment:
continuation line under-indented for visual indent
```suggestion
                                                   api_version='2023-05-01')
```